tests/test_scenario/test_link_velocities.py

```diff
@@ -0,0 +1,314 @@
+
+    else:
+
+        orientation = model.baseOrientation
+        bodyAngularVelocity = model.baseBodyAngularVelocity
+        worldAngularVelocity = model.baseWorldAngularVelocity
+
+    skew = lambda matrix: (matrix - matrix.transpose()) / 2
+    vee = lambda matrix: [matrix[2, 1], matrix[0,2], matrix[1, 0]]
+
+    for _ in range(int(0.5 / dt)):
+
+        W_R_L_old = to_matrix(orientation())
+        assert gazebo.run()
+        W_R_L_new = to_matrix(orientation())
+
+        dot_rotation_matrix = (W_R_L_new - W_R_L_old) / dt
+        world_velocity = dot_rotation_matrix @ W_R_L_new.transpose()
+
+        # Test the world velocity (MIXED)
+        assert vee(skew(world_velocity)) == pytest.approx(worldAngularVelocity(),
+                                                          abs=0.005)
+
+        # Test the BODY velocity
+        body_velocity = W_R_L_new.transpose() @ dot_rotation_matrix
+        assert vee(skew(body_velocity)) == pytest.approx(bodyAngularVelocity(),
+                                                         abs=0.005)
+
+
+# 3. ACCELERATION LINEAR
+@pytest.mark.parametrize("default_world, get_model, link_name", [
+                          ((1.0 / 10_000, 1.0, 1), get_random_panda, "panda_link7"),
+                          # ((1.0 / 10_000, 1.0, 1), get_cube, "cube"),  # TODO
+                          ],
+                         indirect=["default_world"])
+def test_linear_acceleration(
+        default_world: Tuple[bindings.GazeboSimulator, bindings.World],
+        get_model: Callable[[bindings.GazeboSimulator, bindings.World], bindings.Model],
+        link_name: str):
+
+    # Get the simulator and the world
+    gazebo, world = default_world
+    dt = gazebo.stepSize()
+
+    # Get the model
+    model = get_model(gazebo, world)
+
+    # Get the link.
+    # If the link is the base link, get the data through the base methods.
+    link = model.getLink(link_name)
+
+    if link.name() != model.baseFrame():
+
+        orientation = link.orientation
+        worldLinearVelocity = link.worldLinearVelocity
+        bodyLinearAcceleration = link.bodyLinearAcceleration
+        worldLinearAcceleration = link.worldLinearAcceleration
+
+    else:
+
+        orientation = model.baseOrientation
+        worldLinearVelocity = model.baseWorldLinearVelocity
+        bodyLinearAcceleration = model.baseBodyLinearAcceleration
+        worldLinearAcceleration = model.baseWordLinearAcceleration
+
+    # 0.5 seconds of simulation
+    for _ in range(int(0.5 / dt)):
+
+        velocity_old = np.array(worldLinearVelocity())
+        assert gazebo.run()
+        velocity_new = np.array(worldLinearVelocity())
+
+        world_acceleration = (velocity_new - velocity_old) / dt
+
+        # By time to time there are steps where the acceleration becomes extremely high,
+        # like 1000 m/s/s when the average is never exceeds 4 m/s/s.
+        # We exclude those points. We should understand why this happens.
```

Comment:
😨

Reply:
Not sure what's happening there :thinking:

Reply:
There are only few cases in `0.5 sec * 10kH = 5k runs`